src/shared/install.c

```diff
@@ -1413,6 +1416,38 @@ static int unit_file_search(
 
         assert(info->name);
 
+        if (unit_name_is_valid(info->name, UNIT_NAME_INSTANCE)) {
+                r = unit_name_template(info->name, &template);
+                if (r < 0)
+                        return r;
+        }
+
+        STRV_FOREACH(p, paths->search_path) {
+                _cleanup_free_ char *path = NULL;
+
+                path = strjoin(*p, "/", info->name, ".d", NULL);
+                if (!path)
+                        return -ENOMEM;
+                strv_extend(&dirs, path);
```

Comment:
We have `strv_consume` for this case, and then _cleanup_free_ is not needed. Also, the return value must be checked.

Reply:
Got it. Thank you.